libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -157,9 +157,8 @@ bool StaticAnalyzer::visit(VariableDeclaration const& _variable)
 	}
 	else if (_variable.isStateVariable())
 	{
-		set<StructDefinition const*> structsSeen;
-		TypeSet oversizedSubTypes;
-		if (structureSizeEstimate(*_variable.type(), structsSeen, oversizedSubTypes) >= bigint(1) << 64)
+		vector<Type const*> oversizedSubtypes;
+		if (storageSizeUpperBound(*_variable.type(), &oversizedSubtypes) >= bigint(1) << 64)
 			m_errorReporter.warning(
```

Comment:
We report collision-warnings in **StaticAnalyzer**, while oversized-object-errors are handled in **TypeChecker**. Should we do both in **TypeChecker**?

Reply:
If it can be done early, it should be done early.